packages/deployment-service/src/functions/pipeline/version-deploy.ts

```diff
@@ -20,10 +20,26 @@ export const versionDeploy: SQSHandler = async (event: SQSEvent, context: Contex
       const deployTaskIndex = pipelineRunner.tasks?.findIndex((task) => task.functionName === 'DEPLOY')
 
       // TODO check status
-      if (deployTaskIndex === -1 || typeof deployTaskIndex === 'undefined') {
+      if (deployTaskIndex === -1 || typeof deployTaskIndex === 'undefined' || !pipelineRunner.tasks) {
```

Comment:
Can you not do `if (!deployTaskIndex || !pipelineRunner.tasks) {` ?

Reply:
type checking as deployTaskIndex is a return from findIndex resulting in half spaceship result and possibly 'undefined' types. This condition only accepts 0 and above and doesn't treat 0 as false boolean.